014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/dbpool/AgroalPoolTest.java

```diff
@@ -0,0 +1,188 @@
+    String idleSec;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.max-size")
+    int datasourceMaxSize;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.min-size")
+    int datasourceMinSize;
+
+    @Test
+    @Order(1)
+    public void idleTimeoutTest() throws InterruptedException {
+        makeApplicationQuery();
+        Thread.sleep(3000);
+        assertEquals(1, activeConnections(), "agroalCheckIdleTimeout: Expected " + datasourceMinSize + " active connections");
+    }
+
+    @Test
+    @Order(2)
+    public void poolTurnoverTest() throws InterruptedException {
+        final int events = 2500;
+        final AtomicInteger max = new AtomicInteger(0);
+        CountDownLatch queuedEvents = new CountDownLatch(events);
+
+        Multi<Tuple2<Long, Long>> combined = Multi.createBy().combining()
+                .streams(makeApplicationQueryAsync(events), activeConnectionsAsync(events)).asTuple();
+
+        combined.subscribe().with(t -> {
+            Long activeCon = t.getItem2();
+            if (max.intValue() < activeCon) {
+                max.set(activeCon.intValue());
+            }
+            assertTrue(datasourceMaxSize >= activeCon, "More mysql SQL connections than pool max-size");
+            assertTrue(datasourceMinSize <= activeCon, "Less mysql SQL connections than pool min-size");
+            queuedEvents.countDown();
+        });
+
+        LOG.info("agroalCheckDbPoolTurnover: Max connection requested: " + max);
+        queuedEvents.await(TIMEOUT_SEC, TimeUnit.SECONDS);
+        assertEquals(0, queuedEvents.getCount(),
+                String.format("agroalCheckDbPoolTurnover: Missing %d events.", queuedEvents.getCount()));
+    }
+
+    @RepeatedTest(10)
+    @Order(3)
+    public void borderConditionBetweenIdleAndGetConnectionTest() throws InterruptedException {
+        final int concurrentRequestAmount = 2;
+        CountDownLatch queuedEvents = new CountDownLatch(concurrentRequestAmount);
+        Multi.createFrom().range(0, concurrentRequestAmount).subscribe().with(n -> Multi.createFrom().ticks()
+                .every(Duration.ofSeconds(getIdle() + 3))
+                .onOverflow().drop()
+                .onFailure().invoke(e -> Assertions.fail("Unexpected exception " + e.getMessage()))
+                .subscribe().with(i -> {
+                    makeApplicationQuery();
+                    queuedEvents.countDown();
+                }));
+
+        queuedEvents.await(TIMEOUT_SEC, TimeUnit.SECONDS);
+        assertEquals(0, queuedEvents.getCount(),
+                String.format("agroalCheckBorderConditionBetweenIdleAndGetConnection: Missing %d events.", queuedEvents.getCount()));
+    }
+
+    @RepeatedTest(100)
+    @Order(4)
+    public void concurrentLoadTest() {
+        final int concurrentRequestAmount = 20;
+        Multi.createFrom().range(0, concurrentRequestAmount).subscribe()
+                .with(n -> LOG.info("agroalConcurrentLoadTest: Result : " + users.count()));
```

Comment:
This test depends on the execution of other tests which is generally a bad practice. Can we assert this condition as part of the other code instead?

Reply:
All tests are an isolated tests.